Post on Go Contrib modules removal (code owners needed)

theletterf opened the pull request:
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/4502

dmathieu commented on content/en/blog/2024/go-contrib-removal.md:
```diff
@@ -0,0 +1,49 @@
+[open an issue](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/new?assignees=&labels=&projects=&template=owner.md&title=).
+
+The only requirements is that you're a member of the OpenTelemetry organization
+and have a good working knowledge of the code you seek to maintain.
```
We also ask folks to have a good working knowledge of the code they wish to own.
Maybe this section should link to https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/CONTRIBUTING.md#code-owners?

theletterf replied:
I think that's covered by 

> have a good working knowledge of the code you seek to maintain. 

Guidelines are linked in line 36.

pellared commented on content/en/blog/2024/go-contrib-removal.md:
```diff
@@ -0,0 +1,50 @@
+the removal is planned for no sooner than August 21, 2024, unless a code owner
+is found. The list includes the following modules:
+
+- `go.opentelemetry.io/otel/detectors/aws/ec2`
```
Can we add hyperlinks to https://pkg.go.dev? Example:

```suggestion
- [`go.opentelemetry.io/otel/detectors/aws/ec2`](https://pkg.go.dev/go.opentelemetry.io/otel/detectors/aws/ec2)
```

Same comment applies for other modules

theletterf replied:
Applying now!

theletterf replied:
@pellared @dmathieu The first four modules seem to be hosted on contrib, too, not in the code SDK. Could you confirm?

pellared replied:
You are right.

pellared reviewed (approved):
Just one comment 👍

cartermp commented on content/en/blog/2024/go-contrib-removal.md:
```diff
@@ -0,0 +1,50 @@
+cSpell:ignore: aws Benedetti Fabrizio Ferri gopkg labstack macaron moduled otelaws otelecho otellambda otelmacaron otelmongo otelmux
+---
+
+The Go SIG will remove Contrib modules that lack code owners. For each module,
```
I think it's worth clarifying exactly what removal means and if the code will keep working for people. This will likely be top of mind for anyone.

theletterf replied:
Excellent point. @pellared What's the fate of the "removed" modules?

pellared replied:
Existing versions are still available (and not retracted). "Removed" means that no new versions will be published until a code owner is found.

chalin replied:
In that case, "removed" isn't a good term, ... unless that's actually what you have in mind in the long term.

Ah, based on https://github.com/open-telemetry/opentelemetry-go-contrib/issues/5542 for example, it seems that it is your intention to have a two-phased approach for modules w/o code owners:

- Deprecate
- Remove

If that's the case, then this post should reflect that. Also, to address the concerns raised by @cartermp, timeframes should be given for each  phase/stage.

dmathieu replied:
The code will be removed from the repositories. But published packages/releases will remain available.

chalin commented on content/en/blog/2024/go-contrib-removal.md:
```diff
@@ -0,0 +1,50 @@
+cSpell:ignore: aws Benedetti Fabrizio Ferri gopkg labstack macaron moduled otelaws otelecho otellambda otelmacaron otelmongo otelmux
+---
+
+The Go SIG will remove Contrib modules that lack code owners. For each module,
+the removal is planned for no sooner than August 21, 2024, unless a code owner
+is found. The list includes the following modules:
```
Two comments here:

- Maybe clarify that such modules are considered deprecated as of the date of this post
  - If this is the case, will they all be marked as deprecated by the SIG, e.g. https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5636?
- August 21 seems like an arbitrary day. Could it be EOM or maybe Sept 1? (See the suggestion below.)

Also, maybe this could be rephrased to read something like: 

```suggestion
The Go SIG will remove contrib modules that lack code owners starting September 1, 2024.

Currently unowned modules include the following:
```

theletterf replied:
@pellared @dmathieu WDYT about the the above questions?

pellared replied:
> If this is the case, will they all be marked as deprecated by the SIG,

Yes.

> August 21 seems like an arbitrary day

Let's please keep this date as we already have this date in a lot of places and change it will be a PITA 😉

chalin commented on content/en/blog/2024/go-contrib-removal.md:
```diff
@@ -0,0 +1,50 @@
+
+## I want to become a code owner! What do I do?
+
+To become a code owner, you just need to
+[open an issue](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/new?assignees=&labels=&projects=&template=owner.md&title=).
+
+The only requirements is that you're a member of the OpenTelemetry organization
+and have a good working knowledge of the code you seek to maintain.
```
This needs some rework: the prerequisites need to be specified first, then a member can be directed to "just" open an issue.

theletterf replied:
Yup!

theletterf commented:
@pellared @dmathieu @chalin PTAL, I've addressed all comments (I think). I'd rather keep the clickbaity title, though — it's hard to find volunteers these days, and "removal" is quite true in the end.

chalin reviewed (commented):
I'm ok with the title: it drives the most important point home.

On the other hand, @pellared seemed to indicate that the modules are considered deprecated -- see https://github.com/open-telemetry/opentelemetry.io/pull/4543#discussion_r1616714486. Can we add a comment to that effect?

theletterf commented on content/en/blog/2024/go-contrib-removal.md:
```diff
@@ -0,0 +1,52 @@
+
+## I want to become a code owner! What do I do?
+
+To become a code owner of one of the modules, you need to be a member of the
```
+1 let me add that.

theletterf commented:
@chalin Addressed your comments in https://github.com/open-telemetry/opentelemetry.io/pull/4543/commits/08a1098331c7beed1b06d5eb11c33c1e96a5ca4c and https://github.com/open-telemetry/opentelemetry.io/pull/4543/commits/93967b36fd867edbe0912e0b9a7efbada043e722 — thanks!
